ButterSample/ButterSample/ViewController.swift

```diff
@@ -14,46 +14,32 @@ class ViewController: UIViewController {
         super.viewDidLoad()
     }
     
-    func makeRequest(at endpoint: Endpoint) {
+	func makeRequest<T: Decodable>(responseType: T.Type, at endpoint: Endpoint) {
         let router = Router()
         spinner.startAnimating()
-        router.makeRequest(endpoint: endpoint) { result in
+		let decoder = JSONDecoder()
+		decoder.dateDecodingStrategy = .secondsSince1970
+		router.makeRequest(responseType: responseType, endpoint: endpoint, decoder: decoder) { result in
```

Comment:
Something we did at nerdery was make the json decoder / encoder a component of the request object (what you've named as your `Endpoint` - reason being that decoding strategies might very well change between different requests, for would be the same for any particular endpoint

Reply:
i like this idea, i'll update

Reply:
updated in f666fef